### PR TITLE
[OPMONDEV-171]: code clean up

### PR DESCRIPTION
**Pull request by wisecrow**
1. Move `api.views.py` helpers to `api.helpers.py`
2. Add missing types

### CODE REVIEW

File: opendata_module/opmon_opendata/api/helpers.py [hunk 1]
```diff
@@ -0,0 +1,213 @@
+import json
+from datetime import datetime
+from gzip import GzipFile
+from io import BytesIO
+from typing import Generator, List, Optional, Tuple, TypedDict
+
+import psycopg2
+from dateutil import relativedelta
+from django.core.handlers.wsgi import WSGIRequest
```
Comment (justinask7) [inline opendata_module/opmon_opendata/api/helpers.py:9]: Import order seems off... Should psycopg2 be at the bottom of this paragraph?

File: opendata_module/opmon_opendata/api/helpers.py [hunk 2]
```diff
@@ -0,0 +1,213 @@
+import json
+from datetime import datetime
+from gzip import GzipFile
+from io import BytesIO
+from typing import Generator, List, Optional, Tuple, TypedDict
```
Comment (justinask7) [inline opendata_module/opmon_opendata/api/helpers.py:5]:
```suggestion
from datetime import datetime
from gzip import GzipFile
from io import BytesIO
import json
from typing import Generator, List, Optional, Tuple, TypedDict
```

File: opendata_module/opmon_opendata/api/helpers.py [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+    date_columns = [column_name for column_name, column_type in column_names_and_types
+                    if column_type == 'date' and column_name in columns]
+    data_cursor = postgres.get_data_cursor(
+        constraints=constraints, columns=columns, order_by=order_clauses, limit=limit)
+
+    return data_cursor, columns, date_columns
+
+
+def generate_meta_file(postgres: PostgreSQL_Manager, columns: List[str],
+                       constraints: List[dict], order_clauses: List[dict], field_descriptions: dict) -> bytes:
```
Comment (justinask7) [inline opendata_module/opmon_opendata/api/helpers.py:82]:
```suggestion
                       constraints: list[dict], order_clauses: list[dict], field_descriptions: dict) -> bytes:
```
Right?

File: opendata_module/opmon_opendata/api/helpers.py [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+
+class OrderByType(TypedDict):
+    column: str
+    order: str
+
+
+def generate_ndjson_stream(postgres: PostgreSQL_Manager, date: datetime,
+                           columns: List[str], constraints: List[dict],
+                           order_clauses: List[dict], settings: dict) -> Generator[bytes, None, None]:
+    """Creates a gzipped ndjson file iterator suitable for StreamingHttpResponse."""
```
Comment (justinask7) [inline opendata_module/opmon_opendata/api/helpers.py:25]:
```suggestion
    """Creates a gzipped json file iterator suitable for StreamingHttpResponse."""
```

**Review by justinask7 — APPROVED** [review]

File: opendata_module/opmon_opendata/api/helpers.py (hunk 4 above)
Comment (wisecrow) [reply opendata_module/opmon_opendata/api/helpers.py:25]: http://ndjson.org format is used here. Technically is it json, but could be worth noting the format.

File: opendata_module/opmon_opendata/api/helpers.py (hunk 3 above)
Comment (wisecrow) [reply opendata_module/opmon_opendata/api/helpers.py:82]: @justinask7 Project uses python3.8. at this moment. Changes you suggest were introduced in python3.9

Comment (justinask7) [reply opendata_module/opmon_opendata/api/helpers.py:82]: But I see `dict` used here then 🤔

Comment (wisecrow) [reply opendata_module/opmon_opendata/api/helpers.py:82]: Yes, this is correct: using generics as types is supported in python3.8. but you can not use parameters for generics as this feature was introduced with PEP-585. You need to use `typing.List` for that.

File: opendata_module/opmon_opendata/api/helpers.py (hunk 2 above)
Comment (wisecrow) [reply opendata_module/opmon_opendata/api/helpers.py:5]: Good catch! I used https://pypi.org/project/isort/ to sort imports, but strangely, this util does not correct import order.

**Pull request merged**